src/js/angular/repositories/controllers.js

```diff
@@ -469,12 +485,24 @@ function AddRepositoryCtrl($scope, toastr, $repositories, $location, Upload, isE
         });
     };
 
-    if ($scope.isEnterprise) {
-        $scope.getConfig('worker');
-    } else if ($scope.isFreeEdition) {
-        $scope.getConfig('free');
+    $scope.isRepositoryTypeValid = function () {
+        if ($scope.isEnterprise && ($scope.repositoryType === REPOSITOTY_TYPES.eeMaster
+            || $scope.repositoryType === REPOSITOTY_TYPES.eeWorker)) {
+            return true;
+        } else if (!$scope.isFreeEdition && !$scope.isEnterprise
+            && $scope.repositoryType === REPOSITOTY_TYPES.se) {
+            return true;
+        } else {
+            return ($scope.isFreeEdition && $scope.repositoryType === REPOSITOTY_TYPES.free)
+                || $scope.repositoryType === REPOSITOTY_TYPES.ontop;
+        }
+    };
+
+    if ($scope.repositoryType && $scope.isRepositoryTypeValid()) {
```

Comment:
I'd prefer such functions to accept parameter. In this case the repositoryType. This would make it more explicit and "readable"

Reply:
Done